plugins/techdocs/src/reader/components/TechDocsProgressBar.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { Progress } from '@backstage/core';
+
+const TechDocsProgressBar = () => {
+  const [hasBeenDelayed, setHasBeenDelayed] = useState(false);
+
+  const delayReason = `Docs are still loading...Backstage takes some extra time to load docs
+  for the first time. The subsequent loads are much faster.`;
+
+  // Allowed time that docs can take to load (in milliseconds)
+  const allowedDelayTime = 5000;
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setHasBeenDelayed(true);
+    }, allowedDelayTime);
```

Comment:
This might be simpler?
```suggestion
    const timer = setTimeout(() => setHasBeenDelayed(true), allowedDelayTime);
```

Reply:
Please use something like useMountedState in all cases of delayed state modification, otherwise you will get console warnings every time an update happens after unmount https://github.com/streamich/react-use/blob/master/docs/useMountedState.md#usage

Reply:
Aha! Great advice @freben, thank you so much. 🙇

Reply:
@freben I've updated the code with `useMountedState`. @OrkoHunter I couldn't find any test written for the main `Progress.tsx` to refer. How should I go about it?

Reply:
@themousepotato We have tests for [other components](https://github.com/spotify/backstage/tree/master/plugins/techdocs/src/reader/components) in TechDocs to take inspiration from.

In my thinking, we can test that the component renders, and after X seconds it has some text.

Reply:
Just make sure to use fake timers instead of actual wall-clock waiting which slows down tests :) https://jestjs.io/docs/en/22.x/timer-mocks